ipn/ipnlocal/local.go

```diff
@@ -3238,6 +3236,17 @@ func (b *LocalBackend) requestEngineStatusAndWait() {
 	b.statusLock.Unlock()
 }
 
+func (b *LocalBackend) resetControlClientLocked() {
+	if b.cc == nil {
+		return
+	}
+	// Call Shutdown in a goroutine, as it may require
+	// locks that we're holding. We do not need to wait
```

Comment:
Is it locks that we're holding? Based on the stack that I observed, the deadlock was contained with the control client itself.

Reply:
I thought it was a lock but I could be wrong, reworded regardless.